### PR TITLE
[BISERVER-12841] Fixed issue with password not taking on recreated PU…

### PR DESCRIPTION
**Pull request by angel-ramoscardona**
…C user

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 28m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1896**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4686/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java [hunk 1]
```diff
@@ -444,9 +444,11 @@ public void deleteUser( Session session, final IPentahoUser user ) throws NotFou
         while ( currentGroups.hasNext() ) {
           currentGroups.next().removeMember( jackrabbitUser );
```
Comment (NJtwentyone) [inline repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java:445]: update header file date

**Comment by NJtwentyone** [comment]
other than the 1 comment. all seems good.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 29m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl repository
```
---
:ok_hand: All tests passed!
 
Tests run: **1896**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/platform/job/PR-4686/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java [hunk 2]
```diff
@@ -444,9 +444,11 @@ public void deleteUser( Session session, final IPentahoUser user ) throws NotFou
...
         }
+        getUserCache().remove( jackrabbitUser.getID() );
         purgeUserFromCache( user.getUsername() );
```
Comment (pentaho-lmartins) [inline repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java:448]: There was already a call to remove the user from the cache. Now we're removing it twice?

File: repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java [hunk 3]
```diff
@@ -444,9 +444,11 @@ public void deleteUser( Session session, final IPentahoUser user ) throws NotFou
...
         // [BISERVER-9215]
         jackrabbitUser.remove();
+        session.save();
```
Comment (pentaho-lmartins) [inline repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java:451]: Maybe it was just the save call that was necessary?

File: repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java (hunk 2 above)
Comment (angel-ramoscardona) [reply repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java:448]: The purgeUserFromCache() method name is a misnomer, that method only removes the user from the userDetailsCahche and not the userCache. I thought of modifying the method to clear both but you need the jackrabit user ID to remove the user from the userCache and some of the places where this method is called from don't have the jackrabit user available. Also, I didn't know if the places where they called this method onle needed to remove the user from both caches or would I break something, that is why I added the userCache remove outside the misnamed method.

File: repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java (hunk 3 above)
Comment (angel-ramoscardona) [reply repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java:451]: The save call is definitively required. But I think the other change is required as well. See my comment on it.

**Pull request merged**